components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -348,6 +360,13 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
             // set the token request context to be used by downstream handlers. This is introduced as a fix for
             // IDENTITY-4111.
             OAuth2Util.setTokenRequestContext(tokReqMsgCtx);
+
+            AuthenticatedUser authorizedUser = tokReqMsgCtx.getAuthorizedUser();
+            if (authorizedUser.getAuthenticatedSubjectIdentifier() == null) {
+                authorizedUser.setAuthenticatedSubjectIdentifier(
```

Comment:
Will there be a performance impact for grant types like client_credentials which usually do not load the SP to determine the sub value?

Reply:
A suggestion from me is to load only the part of the SP related to subject attribute instead of the whole SP even in the worst case.